hilic-data-viewer/src/App.js

```diff
@@ -203,15 +208,19 @@ class App extends Component {
    */
   getCharge(peptide){
     // Check if the peptide is undefined before proceeding
-    if(peptide === undefined)
-      return 0;
+    if(peptide === undefined || peptide === "")
+      return undefined;
 
-  
+    const aminoAcids = "ARNDCQEGHILKMFPSTWYV"; // all valid amino acids
     const charged = "KRH"; // amino acids which affect the charge of the peptide
-    var charge = 0; // charge of the peptide
+    var charge = 1; // charge of the peptide (accounting for end terminus)
 
     // Calculate the charge of the peptide
     for(var acid of peptide){
+      if(!aminoAcids.includes(acid)){
+        // Return undefined if the peptide contains an invalid amino acid
+        return undefined;
+      }
       if(charged.includes(acid)){
```

Comment:
i would do an `if(aminoAcids.includes(acid) { blah; } else { return undefined; }` to save performance. One check per loop rather than two checks per loop